### PR TITLE
Improve the CI for the fmt

### PR DESCRIPTION
**Pull request by TeoDev1611**
Hi I improve the CI for the lua files with stylua actions.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Format Checker
+
+on:
+    push:
+        branches: [main]
+    pull_request:
+        branches: [main]
```
Comment (williamboman) [inline .github/workflows/ci.yml:7]: Shouldn't this preferably run on all pull requests, and not only the main branch? Or does this mean it'll only run on PRs that targets the main branch?

File: .styluaignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+doc/
+plugin/
+.editorconfig
+*.md
```
Comment (williamboman) [inline .styluaignore:4]: Is this file really needed? Will the GitHub action try to format these files?

**Review by williamboman — APPROVED** [review]
Cool!!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TeoDev1611) [reply .github/workflows/ci.yml:7]: Yeah god idea i fix this now

File: .styluaignore (hunk 2 above)
Comment (TeoDev1611) [reply .styluaignore:4]: Is for prevent stylua format other folder and this close and show a exit code but i try test without this file

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TeoDev1611) [reply .github/workflows/ci.yml]: Done

**Review by williamboman — APPROVED** [review]

**Comment by williamboman** [comment]
🚀
